Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -56,25 +56,81 @@ void
 WarpX::UpdatePlasmaInjectionPosition (amrex::Real a_dt)
 {
     const int dir = moving_window_dir;
-    // Continuously inject plasma in new cells (by default only on level 0)
-    if (WarpX::warpx_do_continuous_injection and (WarpX::gamma_boost > 1)){
-        // In boosted-frame simulations, the plasma has moved since the last
-        // call to this function, and injection position needs to be updated
-        current_injection_position -= WarpX::beta_boost *
+
+    // Loop over species
+    const int n_species = mypc->nSpecies();
+    for (int i=0; i<n_species; i++)
+    {
+        WarpXParticleContainer& pc = mypc->GetParticleContainer(i);
+
+        // Continuously inject plasma in new cells (by default only on level 0)
+        if (pc.doContinuousInjection())
+        {
+            PlasmaInjector* plasma_injector = pc.GetPlasmaInjector();
+            if (plasma_injector == nullptr) return;
```

Comment:
Should we use `continue` here instead of `return`, so that we go to the next species in the `for` loop?

Reply:
Thanks, I agree, I think this would be a bug. Fixed in [d2af9e9](https://github.com/ECP-WarpX/WarpX/pull/3958/commits/d2af9e9cadb7b8deed5a16c28835dcf30b24dfa0).